stdlib/lists.gr

```diff
@@ -87,3 +91,81 @@ let forEachi = (f, l) => {
   }
   help(f, l, 0)
 }
+
+let rec filter = (f, l) => {
+  match (l) {
+    | [] => []
+    | [hd, ...tl] => if (f(hd)) [hd, ...filter(f, tl)] else filter(f, tl)
+  }
+}
+
+let rec reject = (f, l) => {
+  match (l) {
+    | [] => []
+    | [hd, ...tl] => if (!f(hd)) [hd, ...reject(f, tl)] else reject(f, tl)
+  }
+}
+
+let hd = (l) => {
+  match (l) {
+    | [] => fail 'hd'
+    | [hd, ...tl] => hd
+  }
+}
+
+let tl = (l) => {
+  match (l) {
+    | [] => fail 'tl'
+    | [hd, ...tl] => tl
+  }
+}
+
+let rec nth = (n, l) => {
+  if (n == 0) hd(l) else nth(n - 1, tl(l))
+}
```

Comment:
This should probably `fail 'nth'` if `n` is greater than the length of the list rather than `fail 'hd'`. I don't think that `nth` needs to support negative indexes, so we should `fail` in that case too. Once exceptions are merged we can change it to `raise InvalidArgument`.